apps/extension/src/utils/index.ts

```diff
@@ -170,14 +170,24 @@ export const parseTransferType = (
  * @returns Account type for public API
  */
 export const toPublicAccount = (derivedAccount: DerivedAccount): Account => {
-  const { alias, address, type, publicKey, owner, pseudoExtendedKey } =
-    derivedAccount;
+  const {
+    alias,
+    address,
+    type,
+    publicKey,
+    owner,
+    pseudoExtendedKey,
+    source,
+    timestamp,
+  } = derivedAccount;
   const isShielded = type === AccountType.ShieldedKeys;
   const account: Account = {
     alias,
     address,
     type,
     pseudoExtendedKey,
+    source,
+    timestamp,
   };
   if (isShielded) {
     account.viewingKey = owner;
```

Comment:
Once thought: Move `pseudoExtendedKey` (and maybe `timestamp` if it's required in Namadillo) down here, so we don't return `undefined` values in transparent accounts. If we don't need `source` in Namadillo, I would vote to drop it from here. Maybe timestamp can be a shielded-only thing too?